website/src/components/Chat/PluginsChooser.tsx

```diff
@@ -0,0 +1,257 @@
+      newPlugins.forEach((plugin) => (plugin.enabled = false));
+      newPlugins[selectedPluginIndex!].enabled = true;
+      newPlugins[selectedPluginIndex!].plugin_config = plugin.plugin_config;
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      //onPluginActivated(newPlugins[selectedPluginIndex!].enabled);
+    }
+    onClose();
+  }, [onClose, control, plugins, selectedPluginIndex]);
+
+  const handlePluginAdd = useCallback(() => {
+    const newPlugin: PluginEntry = {
+      url: "",
+      enabled: false,
+    };
+    setValue("plugins", [...plugins, newPlugin]);
+    handlePluginEdit(plugins.length);
+  }, [setValue, plugins, handlePluginEdit]);
+
+  const limitPluginNameLength = (name: string) => {
+    if (!name) {
+      return "";
+    }
+    if (name.length > 15) {
+      return name.substring(0, 15) + "...";
+    }
+    return name;
+  };
+
+  return (
+    <FormControl mb="40px" ref={containerRef}>
+      <Box position="fixed" zIndex="1">
+        <Menu placement="auto">
+          <MenuButton as={Button} pr="3" rightIcon={<Cog />} w={measureDivWidth(containerRef.current) + 15} size="lg">
+            {plugins && plugins[findSelectedPluginIndex()]?.enabled ? (
+              <Box display="flex" justifyContent="flex-start" gap={2}>
+                <Image
+                  src={plugins[findSelectedPluginIndex()]?.plugin_config?.logo_url}
+                  alt="Plugins"
+                  width="25px"
+                  height="25px"
+                  marginRight="5px"
+                />
+                <Text mt="4px" fontSize="15px">
+                  {limitPluginNameLength(plugins[findSelectedPluginIndex()]?.plugin_config?.name_for_human)}
+                </Text>
+              </Box>
+            ) : (
+              <Box display="flex" justifyContent="center" gap={2}>
+                <AttachmentIcon boxSize="5" />
+                {t("plugins")}
+              </Box>
+            )}
+          </MenuButton>
+          <MenuList left="-150px" position="absolute" w="280px">
+            <RadioGroup value={plugins?.findIndex((plugin) => plugin.enabled).toString()}>
+              {plugins?.map((plugin, index) => (
+                <MenuItem key={index}>
+                  <Radio value={index.toString()} onClick={() => handlePluginSelect(index)}>
+                    <Box
+                      display="flex"
+                      justifyContent="flex-start"
+                      gap={2}
+                      mr="4"
+                      onClick={() => handlePluginSelect(index)}
+                    >
+                      <Image
+                        src={plugins[index]?.plugin_config?.logo_url}
+                        alt="Plugins"
+                        width="25px"
+                        height="25px"
+                        marginRight="5px"
+                      />
+                      <Box position="absolute" top="-2px" left="33px">
```

Comment:
and here

why is all this hard-coded positioning necessary? flex-box is not enough?

Reply:
I don't know, it didn't look good without it... i guess